pkg/kudoctl/kudoinit/types.go

```diff
@@ -13,7 +13,16 @@ const (
 )
 
 type InitStep interface {
+	// Returns a one-line description to print
+	Description() string
+
+	// Should return an error if the installation will not be possible
+	PreInstallCheck(client *kube.Client) Result
+
+	// Executes the actual installation
 	Install(client *kube.Client) error
+
+	// Returns the installed artifacts as yaml manifests
 	AsYamlManifests() ([]string, error)
 	AsArray() []runtime.Object
```

Comment:
It is also very confusing to reason about `kudoinit.InitStep`   It stutters on `init`.

Reply:
Renamed.